src/Internal/Wallet/Cip30Mock.purs

```diff
@@ -155,49 +211,69 @@ mkCip30Mock pKey mSKey = do
             Map.filter
               (flip Array.elem (collateralUtxos <#> unwrap >>> _.output))
               utxos
-        -- Convert to CSL representation and serialize
-        cslUtxos <- traverse (liftEffect <<< convertTransactionUnspentOutput)
-          $ Map.toUnfoldable nonCollateralUtxos <#> \(input /\ output) ->
-              TransactionUnspentOutput { input, output }
-        pure $ (byteArrayToHex <<< unwrap <<< toBytes) <$> cslUtxos
-    , getCollateral: fromAff do
+        let
+          xUtxos = Map.toUnfoldable nonCollateralUtxos <#> \(input /\ output) ->
+            TransactionUnspentOutput { input, output }
+        let
+          amountValue = DSV.convertValue =<< fromBytes =<< hexToCborBytes =<<
+            (uorToMaybe amount)
+        if (not hasEnoughAmount amountValue xUtxos) then pure $ maybeToNullOr
```

Comment:
Strange style, works because of [this](https://github.com/purescript/purescript-prelude/blob/68f8012bc2309d9bf5832cdf7316ad052d586905/src/Data/HeytingAlgebra.purs#L71), didn't know about this. I'm guessing this generates more complicated JS code than `not $ hasEnoughAmount amountValue xUtxos`, as the instances have to be resolved recursively for the function instance.

Reply:
Also I don't think the parens are necessary